apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/AbstractSamhsaMatcher.java

```diff
@@ -420,6 +445,7 @@ protected boolean isSamhsaDiagnosis(CodeableConcept concept) {
     return isSamhsaCoding(
         concept,
         this::isSamhsaIcd9Diagnosis,
+        null,
```

Comment:
Curious about this trend of null references that seem to leave NPE paths in subsequent logic.  Would it be better to continue to process them normally, even if they shouldn't happen from a business perspective, or even return a static true/false, rather than allow a potential NPE issue?

Reply:
IMO if we have an unexpected IcdCode with no properly passed value here, we should probably blow up intentionally (Objects.assertNonNull on the values inside the cases) or default to true and log the error if we want to be less explosive.

Reply:
Thanks for raising this. I agree that an NPE should not escape here although an exception I believe is reasonable. How about a special predicate that just throws?